shell/platform/darwin/macos/framework/Source/FlutterMetalCompositor.mm

```diff
@@ -11,16 +11,20 @@
 namespace flutter {
 
 FlutterMetalCompositor::FlutterMetalCompositor(
-    FlutterViewController* view_controller,
+    FlutterViewProvider* view_provider,
     FlutterPlatformViewController* platform_views_controller,
     id<MTLDevice> mtl_device)
-    : FlutterCompositor(view_controller),
+    : FlutterCompositor(view_provider),
       mtl_device_(mtl_device),
       platform_views_controller_(platform_views_controller) {}
 
 bool FlutterMetalCompositor::CreateBackingStore(const FlutterBackingStoreConfig* config,
                                                 FlutterBackingStore* backing_store_out) {
-  if (!view_controller_) {
+  // TODO(dkwingsmt): This class only supports single-view for now. As more
```

Comment:
Is there an issue for overall multi-window effort for macOS. It will be useful to link here.

Reply:
We only have an overall multiwindow issue. Do you think it's worth opening one for the macOS multi-surface?